HackYourFuture-CPH/kp-toolbox-fp-class21
# FP21-56-Integrate firebase authentication

# Description

Google authentification using Firebase is implemented. 
✅ A user can log in using a Google account 
✅ A user can log out - when logging out, the user is redirected to the main page

# How to test?

In the client .env file add `REACT_APP_FIREBASE_API_KEY=AIzaSyARcKPPTVo1aqOCnH-_QAf8rnt3LyUyArY `
Run the client side and try loggin in and out. To check if the user is added in the Firebase db, check Firebase console.

# Notes 

The auth process is based on the context for further reusability of data obtained. So feel free to get the user info from the data obtained from Firebase. The corresponding variable is stored in the context.  

# Checklist

✅ I have performed a self-review of my own code
✅I have followed the name conventions for CSS Classnames and filenames, Components names and filenames, Style filenames, if you are in doubt check the the project README.MD and here https://github.com/HackYourFuture-CPH/curriculum/blob/master/review/review-checklist.md
✅ I have commented my code, particularly in hard-to-understand areas, if you code was simple enough mark the box anyway
✅ I have made corresponding changes to the documentation, if you code was simple enough mark the box anyway
✅ This PR is ready to be merged and not breaking any other functionality
